R/tax_range_time.R

```diff
@@ -0,0 +1,178 @@
+#' age range (in millions of years) of the fossil occurrence. If the age range
+#' columns are provided as (`character`) values (i.e., stage names), the
+#' function will attempt to extract numeric age data from the
+#' Geological Timescale by linking stage names.
+#' @param name \code{character}. The name of the column you wish to be treated
+#' as the input names (e.g., "species" or "genus").
+#' @param min_ma \code{character}. The name of the column you wish to be treated
+#' as the input minimum age range (e.g., "min_ma").
+#' @param max_ma \code{character}. The name of the column you wish to be treated
+#' as the input maximum age range (e.g., "max_ma").
+#' @param scale \code{character}. Specify the desired geological timescale to
+#' be used "GTS2020" or "GTS2012". This argument is only useful if the supplied
+#' "max_ma" and "min_ma" columns are of class \code{character}.
+#' "GTS2020" is the default.
+#' @param plot \code{logical}. Should a plot of the ranges be generated?
+#' @param return_error \code{logical}. Should a vector of numbers be returned
+#' to flag the rows of the `occdf` that cannot be matched to `character`
+#' interval names? This is only relevant if `max_ma` and `min_ma`
+#' are `character` values.
+#'
+#' @return A \code{dataframe} containing the following columns:
+#' unique taxa (`taxa`), taxa ID (`taxa_id`), first appearance of taxa
+#' (`FAD_ma`), last appearance of taxa (`LAD_ma`), and temporal range duration
+#' (`range_myr`) is returned.
+#'
+#' @details The temporal range(s) of taxa are calculated by extracting all
+#' unique taxa (`name` column) from the input `occdf`, and checking their first
+#' and last appearance. The temporal duration of each taxa is also calculated.
+#'
+#' @section Developer(s):
+#' Lewis A. Jones
+#' @section Reviewer(s):
+#' To be reviewed
+#' @importFrom graphics points
+#' @examples
+#' # Grab internal data
+#' occdf <- tetrapods
+#' # Temporal range
+#' tax_range_time(occdf = occdf, name = "accepted_name", plot = TRUE)
+#'
+#' @export
+tax_range_time <- function(occdf,
+                           name = "name",
+                           min_ma = "min_ma",
+                           max_ma = "max_ma",
+                           scale = "GTS2020",
+                           plot = FALSE,
+                           return_error = FALSE) {
+
+  #=== Handling errors ===
+  if (is.data.frame(occdf) == FALSE) {
+    stop("`occdf` should be a dataframe")
+  }
+
+  if (is.logical(plot) == FALSE) {
+    stop("`plot` should be logical (TRUE/FALSE)")
+  }
+
+  if (class(occdf[, max_ma]) != class(occdf[, min_ma])) {
+    stop("The class of max_ma and min_ma must be the same")
+  }
+
+  if (any(c(name, min_ma, max_ma) %in% colnames(occdf) == FALSE)) {
+    stop("Either `name`, `min_ma`, or `max_ma`, do not exist in `occdf`")
+  }
+
+  if (any(is.na(occdf[, name]))) {
+    stop("The `name` column contains NA values")
```

Comment:
Again, it should maybe be mentioned in the documentation that these need to be filtered out.